cypress/integration/app_spec.ts

```diff
@@ -1,70 +1,124 @@
-  cy.intercept('GET', 'https://freesound.org/apiv2/sounds/462808/similar/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings', {fixture:'similar-sounds.json'})
-  cy.intercept('GET', 'https://freesound.org/data/previews/462/462808_8386274-lq.mp3', {fixture:'sound.mp3'})
-  cy.intercept('GET', 'https://freesound.org/apiv2/search/text/*', {fixture:'search-results.json'})
+  cy.intercept("GET", "https://freesound.org/apiv2/sounds/462808", {
+    fixture: "sound.json",
+  });
+  cy.intercept("GET", "/people/qfox123/packs/26158", { fixture: "sound.json" });
+  cy.intercept("GET", "https://freesound.org/apiv2/packs/26158/", {
+    fixture: "pack.json",
+  });
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/packs/26158/sounds/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings",
+    { fixture: "pack-sounds.json" }
+  );
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/sounds/462808/similar/?fields=id%2Cname%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings",
+    { fixture: "similar-sounds.json" }
+  );
+  cy.intercept(
+    "GET",
+    "https://freesound.org/data/previews/462/462808_8386274-lq.mp3",
+    { fixture: "sound.mp3" }
+  );
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/search/text/?fields=id%2Cname%2Cpreviews%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings&page_size=10&query=hello",
+    {
+      fixture: "search-results.json",
+    }
+  );
+
+  cy.intercept(
+    "GET",
+    "https://freesound.org/apiv2/search/text/?fields=id%2Cname%2Cpreviews%2Cduration%2Cnum_downloads%2Cusername%2Cnum_ratings&page_size=10&page=2&query=hello",
+    {
+      fixture: "search-results-page-2.json",
+    }
+  );
 }
 
 describe("Home Page", () => {
   beforeEach(() => {
-    setupServerMocks()
-  })
+    setupServerMocks();
+  });
 
   it("should load the home page", () => {
     cy.visit("/");
-    cy
-      .get(`[data-e2e-id='SoundList']`)
-      .should('be.visible')
+    cy.get(`[data-e2e-id='SoundList']`).should("be.visible");
+  });
+});
+
+describe("Pagination element functionality", () => {
+  beforeEach(() => {
+    setupServerMocks();
+  });
+
+  it("checks if next page button shows next page of results", () => {
+    cy.visit("/search?q=hello");
+    cy.get(`[data-e2e-id="SoundList-track-name"]`)
+      .first()
+      .should("be.visible")
+      .should("have.text", "hello");
+    cy.get(`[data-e2e-id="SoundList-track-url"]`)
+      .first()
+      .should("be.visible")
+      .should("have.attr", "href", "/sound/123");
+    cy.get(`[data-e2e-id="first-page-button"]`).should("be.visible");
+    cy.get(`[data-e2e-id="next-page-button"]`).should("be.visible").click();
+    cy.get(`[data-e2e-id="SoundList-track-name"]`).should(
+      "have.text",
+      `hello.wavHello #3.MP3Hello.wavHello #2.MP3Me Saying "Hello"Hello Scream Male LoudHello #1.MP3hello.wavhello.wavhello.wav`
```

Comment:
Since the `cy.get` query here returns multiple elements, I think it would make sense to just check the first element

Reply:
I think that's one of the things that was causing the tests to fail.

We do `cy.get(...).first()`, which grabs the first track.

But, the first track grabbed is from the first page of search results. We lose access to that track after clicking on next page button, and when we test the first track on the second page of search results, Cypress complains that we're testing a detached element.

```
CypressError
Timed out retrying: cy.should() failed because this element is detached from the DOM.

<span class="overflow-clip" data-e2e-id="SoundList-track-name">hello</span>

Cypress requires elements be attached in the DOM to interact with them.

The previous command that ran was:

  > cy.get()

This DOM element likely became detached somewhere between the previous and current command.

Common situations why this happens:
  - Your JS framework re-rendered asynchronously
  - Your app code reacted to an event firing and removed the element

You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands.
```


![Screen Shot 2020-12-19 at 11 27 04 AM](https://user-images.githubusercontent.com/43425812/102697768-217fa480-41ed-11eb-9833-d154296e73de.png)

Reply:
I think that's one of the things that was causing the tests to fail.

We do `cy.get(...).first()`, which grabs the first track.

But, the first track grabbed is from the first page of search results. We lose access to that track after clicking on the next page button, and when we test the first track on the second page of search results, Cypress complains that we're testing a detached element.

```
CypressError
Timed out retrying: cy.should() failed because this element is detached from the DOM.

<span class="overflow-clip" data-e2e-id="SoundList-track-name">hello</span>

Cypress requires elements be attached in the DOM to interact with them.

The previous command that ran was:

  > cy.get()

This DOM element likely became detached somewhere between the previous and current command.

Common situations why this happens:
  - Your JS framework re-rendered asynchronously
  - Your app code reacted to an event firing and removed the element

You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands.
```


![Screen Shot 2020-12-19 at 11 27 04 AM](https://user-images.githubusercontent.com/43425812/102697768-217fa480-41ed-11eb-9833-d154296e73de.png)